### DISCUSSION TITLE
Script file for rendering 2D Slice files with specifik min and max values

### DISCUSSION THREAD

**Discussion by Mikkelmhfire** [firemodels/fds#13455] [General]
Hi there,

Im trying to automate the proces where i render images from the FDS simulation regarding different quantities such as VISIBILITY AND TEMPERATURE in 2D slices. I have followed the smokeview guide chapter 9, but cant seem to get the last step in the code to work, which happens to be the part where min and max values are set. Regarding the VISIBILITY slices, i need to set the values on the colorscale between 10 meter and 30 meters. Also the values below 10 meters needs to be greyed out or somehow highlited. 

The code beneath here is where i am now. This code needs to be saved as an .ssf file and then you need to write the following command: smokeview - script FILENAME.ssf FILENAME.smv. This generates pictures of 2D slices for visibility and temperature in 10 s and 20 s into the simulation. Hopefully someone out there can help me solve the last part of the code, so that min and max values are set.

RENDERDIR
 .
XSCENECLIP
 0 -0.005400 0 5.405400
YSCENECLIP
 0 -0.005000 0 5.005000
ZSCENECLIP
 0 -0.005000 0 5.005000
SCENECLIP
 0

LOADSLCF
 QUANTITY='SOOT VISIBILITY' PBY=3.4
SETTIMEVAL
 10
RENDERCLIP
 ALL
RENDERONCE
 SOOT_script_slice_test_10
SETTIMEVAL
 20
RENDERCLIP
 ALL
RENDERONCE
 SOOT_script_slice_test_20

LOADSLCF
 QUANTITY='TEMPERATURE' PBY=3.4
SETTIMEVAL
 10
RENDERCLIP
 ALL
RENDERONCE
 TEMP_script_slice_test_10
SETTIMEVAL
 20
RENDERCLIP
 ALL
RENDERONCE
 TEMP_script_slice_test_20

**Comment by gforney** [comment]
try the following.
1. open the Files/Data/Coloring dialog box. then open the slice tab.
2. select the visibility quantity and then set the bounds you want. repeat
for temperature.
3. once set, click the 'Save settings' button to save an ini file.
4.  close smokeview and open back up and load the visibility and
temperature slice files to verify the min and max bounds on the colorbar
are what you set earlier.

if you would like the ini file to have a different name than casename.ini
(where casename.fds is the name of your input file) say
casename_option.ini, then add this line to your script file at the beginning
LOADINIFILE
  casename_option.ini



On Sun, Sep 22, 2024 at 8:30 AM Mikkelmhfire ***@***.***>
wrote:

> Hi there,
>
> Im trying to automate the proces where i render images from the FDS
> simulation regarding different quantities such as VISIBILITY AND
> TEMPERATURE in 2D slices. I have followed the smokeview guide chapter 9,
> but cant seem to get the last step in the code to work, which happens to be
> the part where min and max values are set. Regarding the VISIBILITY slices,
> i need to set the values on the colorscale between 10 meter and 30 meters.
> Also the values below 10 meters needs to be greyed out or somehow highlited.
>
> The code beneath here is where i am now. This code needs to be saved as an
> .ssf file and then you need to write the following command: smokeview -
> script FILENAME.ssf FILENAME.smv. This generates pictures of 2D slices for
> visibility and temperature in 10 s and 20 s into the simulation. Hopefully
> someone out there can help me solve the last part of the code, so that min
> and max values are set.
>
> RENDERDIR
> .
> XSCENECLIP
> 0 -0.005400 0 5.405400
> YSCENECLIP
> 0 -0.005000 0 5.005000
> ZSCENECLIP
> 0 -0.005000 0 5.005000
> SCENECLIP
> 0
>
> LOADSLCF
> QUANTITY='SOOT VISIBILITY' PBY=3.4
> SETTIMEVAL
> 10
> RENDERCLIP
> ALL
> RENDERONCE
> SOOT_script_slice_test_10
> SETTIMEVAL
> 20
> RENDERCLIP
> ALL
> RENDERONCE
> SOOT_script_slice_test_20
>
> LOADSLCF
> QUANTITY='TEMPERATURE' PBY=3.4
> SETTIMEVAL
> 10
> RENDERCLIP
> ALL
> RENDERONCE
> TEMP_script_slice_test_10
> SETTIMEVAL
> 20
> RENDERCLIP
> ALL
> RENDERONCE
> TEMP_script_slice_test_20
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firemodels/fds/discussions/13455>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6UCRRK73ZSEHUSKL5KHSDZX2Z55AVCNFSM6AAAAABOUQSB5CVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGIYTOMBQGM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Glenn Forney

**Reply by Mikkelmhfire** [reply]
Hi gforney, 

I tried doing this but it doenst help. Somehow smokeview cant remember even tho i press the save button...

**Reply by gforney** [reply]
Upload your input and your script file and ill take a look

**Reply by Mikkelmhfire** [reply]
Here you go. Thanks! I've tried adding: SETSLICEBOUNDS in the inputfile, but didnt help.

[To_kemiske_reaktioner-Methanol.txt](https://github.com/user-attachments/files/17093444/To_kemiske_reaktioner-Methanol.txt)
[To_kemiske_reaktioner-Methanol_script.txt](https://github.com/user-attachments/files/17093447/To_kemiske_reaktioner-Methanol_script.txt)

**Reply by Mikkelmhfire** [reply]
Hi again,

Suddently it works adding SETSLICEBOUNDS into the ssf file. Last thing i need is to highlight the colers below 10 meters visibility. Any surgestions?

**Comment by gforney** [comment]
1. Open up the Files/Data/Coloring dialog box
2. Click on the Coloring tab
3.  select "Color below min" and "Color above max" checkboxes and select colors you wish to use to highlight these values.
4.  click on Save settings button to save an ini file with these preferences

**Comment by gforney** [comment]
I added two script keywords, HILIGHTMINVALS and HILIGHTMAXVALS . 
HILIGHTMINVALS
 1 r g b

will use the color r g b (each component ranging between 0 and 255) to color values below the min value appearing on the colorbar.  

HILIGHTMINVALS
 0 r g b

will turn this off (ignoring r g and b)

Similarly for HILIGHTMAXVALS.  update your smv repo and rebuild smokeview or use prebuilt smokeviews located here:
https://github.com/gforney/test_bundles/releases/tag/SMOKEVIEW_TEST .  If you try this let me know if you see any problems
